src/agent/agent.c

```diff
@@ -388,87 +386,52 @@ static int agent_method_start_unit(UNUSED sd_bus_message *m, void *userdata, UNU
                 return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
         }
 
-        if (!systemd_request_start(req, start_unit_callback)) {
+        if (!systemd_request_start(req, unit_lifecycle_method_callback)) {
                 return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
         }
 
         return 1;
 }
 
 /*************************************************************************
- ********** org.containers.hirte.internal.Agent.StopUnit ****************
+ ********** org.containers.hirte.internal.Agent.StartUnit ****************
  ************************************************************************/
 
-static int stop_unit_callback(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
-        _cleanup_systemd_request_ SystemdRequest *req = userdata;
-        Agent *agent = req->agent;
-        const char *job_object_path = NULL;
-
-        if (sd_bus_message_is_method_error(m, NULL)) {
-                /* Forward error */
-                return sd_bus_reply_method_error(req->request_message, sd_bus_message_get_error(m));
-        }
-
-        int r = sd_bus_message_read(m, "o", &job_object_path);
-        if (r < 0) {
-                return sd_bus_reply_method_errorf(req->request_message, SD_BUS_ERROR_FAILED, "Internal Error");
-        }
-
-        AgentJobOp *op = req->userdata;
-
-        if (!agent_track_job(
-                            agent,
-                            job_object_path,
-                            agent_job_done,
-                            agent_job_op_ref(op),
-                            (free_func_t) agent_job_op_unref)) {
-                return sd_bus_reply_method_errorf(req->request_message, SD_BUS_ERROR_FAILED, "Internal Error");
-        }
-
-        return sd_bus_reply_method_return(req->request_message, "");
+static int agent_method_start_unit(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        return agent_run_unit_lifecycle_method(m, (Agent *) userdata, "StartUnit");
```

Comment:
The code for `StartUnit` and `StopUnit` is mostly the same - the same applies to the new `RestartUnit` and `ReloadUnit`. Therefore, I extracted it into a dedicated function. Not sure about the naming, though.

Reply:
sounds fine